source/external/imgui/src/CMakeLists.txt

```diff
@@ -14,13 +14,27 @@ target_compile_definitions(imgui
         WIN32_LEAN_AND_MEAN
 )
 
+set(NC_INCLUDE_DIR "${PROJECT_SOURCE_DIR}/include")
+set(NC_SOURCE_DIR "${PROJECT_SOURCE_DIR}/source/engine/")
+set(NC_EXTERNAL_DIR "${PROJECT_SOURCE_DIR}/source/external")
+
+# Include Dirs
 target_include_directories(imgui
+    PUBLIC
+        $<INSTALL_INTERFACE:include/>
+        $<INSTALL_INTERFACE:external/>
+        $<BUILD_INTERFACE:${NC_INCLUDE_DIR}>
```

Comment:
Is this one needed? I feel like the imgui lib should not need to know about the engine includes, although it may have before for the filtered win32 header.